### ISSUE TITLE
When loading datasets by HuggingFace datasets.load_dataset like cifar10, could it be possible to return the dataset without decoding automatically.

### ISSUE THREAD

**Issue by jychen-habana** [huggingface/optimum-habana#236]
### Feature request

When loading datasets by HuggingFace datasets.load_dataset like cifar10, could it be possible to return the dataset without decoding automatically?

### Motivation

According to https://github.com/huggingface/optimum-habana/pull/189, The scale efficiency is about 72.6% for Gaudi2 and 79.4% for Gaudi, we found that the efficiency of Gaudi2 is low because of the data loader, so we intend to implement a data loader (especially for Gaudi2) based on Habana [Media Pipeline](https://docs.habana.ai/en/latest/Media_Pipeline/Pytorch_Resnet_Media_Pipe.html) to do the Decoding, RandomResizedCrop, RandomHorizontalFlip, and Normalize. 
![image](https://github.com/huggingface/optimum-habana/assets/94743284/71c2af9c-988a-46b6-a62e-546e69cfcfb8)

As described in [cifar10 dataset data-fields](https://huggingface.co/datasets/cifar10#data-fields), when accessing the image column: dataset[0]["image"] the image file is automatically decoded this will be executed on CPU. can it just be like the dataset with a root path to the image files and let our self defined dataloader to do the decoding on HPU?

### Your contribution

Implement a Habana media-based data loader

**Comment by jychen-habana** [comment]
> Hi @jychen-habana!
> 
> If you do:
> 
> ```python
> import datasets
> 
> ds = datasets.load_dataset("cifar10")
> ds = ds.cast_column("img", datasets.Image(decode=False))
> ```
> 
> then images won't be automatically decoded. For instance, `ds["train"][0]["img"]` looks like:
> 
> ```
> {'bytes': b'...', 'path': None}
> ```
> 
> `path` and `bytes` can be None, but never both at the same time.
> 
> Does it help?

Thanks! I will give it a try.

**Comment by jychen-habana** [comment]
close since datasets could set decode to False
